src/util/psbt/responsibilities.rs

```diff
@@ -0,0 +1,142 @@
+use hashes::core::fmt::Debug;
+use secp256k1::{Message, Signature};
+use util::bip32::{DerivationPath, ExtendedPrivKey, Fingerprint};
+use util::psbt::PartiallySignedTransaction;
+use {Address, Amount};
+use {Script, TxOut};
+
+/// This trait corresponds to the Creator and Updater responsibility described in
+/// [BIP-174](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki#creator). As noted
+/// there the both functionalities are most often combined as the creator most likely has more meta
+/// information about the transaction they are creating and is thus able to put it into the PSBT
+/// (which otherwise would be a separate update step).
+pub trait PsbtWallet {
+    /// Wallet backend specific errors like database connection errors that aren't captured in
+    /// `PsbtCreationError`.
+    type WalletError: Debug;
+
+    /// Returns a script to which possible change should be sent
+    fn get_change_script(&mut self) -> Result<Script, Self::WalletError>;
+
+    /// Create a transaction that pays a set of outputs, optionally sending change back to a
+    /// specified change script. If no change script is provided the wallet may overpay as it sees
+    /// fit. This can be used to increase privacy, but should be externally limited for safety
+    /// reasons.
+    fn create_transaction(
+        &mut self,
+        outputs: &[TxOut],
+        change_output: Option<Script>,
```

Comment:
Maybe we want to have a different trait that allows this particular privacy optimization to avoid panics/useless error conditions.